In [2]:
import numpy as np
import pandas as pd
from scipy import stats
import statsmodels.api as sm
import matplotlib.pyplot as plt

In [3]:
data = pd.read_excel('Housing.xlsx')

In [26]:
data.head()

House Price  House Size (sq.ft.) State  Number of Rooms  \
0      1116000                 1940    IN                8   
1       860000                 1300    IN                5   
2       818400                 1420    IN                6   
3      1000000                 1680    IN                7   
4       640000                 1270    IN                5   

   Year of Construction  
0                  2002  
1                  1992  
2                  1987  
3                  2000  
4                  1995

#Multivarite Regression

> Indipendent Variables:"House Size", "Numberof Rooms", "Years of Construction"



In [18]:
x= data[['House Size (sq.ft.)', 'Number of Rooms', 'Year of Construction']]
y = data['House Price']

In [19]:
X1= sm.add_constant(x)
reg = sm.OLS(y,X1).fit()
reg.summary()

/usr/local/lib/python3.8/dist-packages/statsmodels/tsa/tsatools.py:142: FutureWarning: In a future version of pandas all arguments of concat except for the argument 'objs' will be keyword-only
  x = pd.concat(x[::order], 1)


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:            House Price   R-squared:                       0.736
Model:                            OLS   Adj. R-squared:                  0.687
Method:                 Least Squares   F-statistic:                     14.90
Date:                Sun, 25 Dec 2022   Prob (F-statistic):           6.82e-05
Time:                        16:25:42   Log-Likelihood:                -258.43
No. Observations:                  20   AIC:                             524.9
Df Residuals:                      16   BIC:                             528.9
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
========================================================================================
                           coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------
const                -9.452e+06    5.4e+06     -1.752      0.099   -2.09e+07    1.99e+06
House Size (sq.ft.)    341.8271    179.666      1.903      0.075     -39.049     722.703
Number of Rooms        1.16e+04   3.74e+04      0.310      0.760   -6.77e+04    9.08e+04
Year of Construction  4863.5761   2697.969      1.803      0.090    -855.862    1.06e+04
==============================================================================
Omnibus:                        2.140   Durbin-Watson:                   1.938
Prob(Omnibus):                  0.343   Jarque-Bera (JB):                1.747
Skew:                          -0.676   Prob(JB):                        0.418
Kurtosis:                       2.484   Cond. No.                     5.40e+05
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 5.4e+05. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

Inferences:


> R^2 0.736 means the model has significant explanatory power.


> P value is greater than 5%. This means this co-effecient are not statistically significant.

>  so we need to run more regression to make a conclusion







Reg-1:  Independent Variables: "House Size(sq.ft.), "Number of Rooms"

In [20]:
x= data[['House Size (sq.ft.)', 'Number of Rooms']]
y = data['House Price']

In [23]:
X1= sm.add_constant(x)
reg = sm.OLS(y,X1).fit()
reg.summary()

/usr/local/lib/python3.8/dist-packages/statsmodels/tsa/tsatools.py:142: FutureWarning: In a future version of pandas all arguments of concat except for the argument 'objs' will be keyword-only
  x = pd.concat(x[::order], 1)


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:            House Price   R-squared:                       0.683
Model:                            OLS   Adj. R-squared:                  0.645
Method:                 Least Squares   F-statistic:                     18.30
Date:                Sun, 25 Dec 2022   Prob (F-statistic):           5.77e-05
Time:                        16:29:00   Log-Likelihood:                -260.28
No. Observations:                  20   AIC:                             526.6
Df Residuals:                      17   BIC:                             529.6
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
=======================================================================================
                          coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------
const                2.737e+05   1.03e+05      2.655      0.017    5.62e+04    4.91e+05
House Size (sq.ft.)   314.1363    190.485      1.649      0.117     -87.752     716.025
Number of Rooms      1.944e+04   3.95e+04      0.492      0.629   -6.39e+04    1.03e+05
==============================================================================
Omnibus:                        1.326   Durbin-Watson:                   1.852
Prob(Omnibus):                  0.515   Jarque-Bera (JB):                0.810
Skew:                          -0.487   Prob(JB):                        0.667
Kurtosis:                       2.853   Cond. No.                     5.89e+03
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 5.89e+03. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

Reg -2_ Independent Variable: "Number of Rooms", "Year of Construction"

In [24]:
x= data[['Year of Construction', 'Number of Rooms']]
y = data['House Price']

X1= sm.add_constant(x)
reg = sm.OLS(y,X1).fit()
reg.summary()

/usr/local/lib/python3.8/dist-packages/statsmodels/tsa/tsatools.py:142: FutureWarning: In a future version of pandas all arguments of concat except for the argument 'objs' will be keyword-only
  x = pd.concat(x[::order], 1)


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:            House Price   R-squared:                       0.677
Model:                            OLS   Adj. R-squared:                  0.639
Method:                 Least Squares   F-statistic:                     17.79
Date:                Sun, 25 Dec 2022   Prob (F-statistic):           6.79e-05
Time:                        16:29:01   Log-Likelihood:                -260.47
No. Observations:                  20   AIC:                             526.9
Df Residuals:                      17   BIC:                             529.9
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
========================================================================================
                           coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------
const                -8.471e+06   5.77e+06     -1.468      0.160   -2.06e+07     3.7e+06
Year of Construction  4424.7160   2887.793      1.532      0.144   -1667.996    1.05e+04
Number of Rooms       7.824e+04    1.4e+04      5.574      0.000    4.86e+04    1.08e+05
==============================================================================
Omnibus:                        2.115   Durbin-Watson:                   1.959
Prob(Omnibus):                  0.347   Jarque-Bera (JB):                1.400
Skew:                          -0.407   Prob(JB):                        0.497
Kurtosis:                       1.991   Cond. No.                     4.34e+05
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 4.34e+05. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

Reg-3- Independent Variable: "House Size(sq.ft.)" , "Year of Construction"

In [25]:
x= data[['Year of Construction', 'House Size (sq.ft.)']]
y = data['House Price']

X1= sm.add_constant(x)
reg = sm.OLS(y,X1).fit()
reg.summary()

/usr/local/lib/python3.8/dist-packages/statsmodels/tsa/tsatools.py:142: FutureWarning: In a future version of pandas all arguments of concat except for the argument 'objs' will be keyword-only
  x = pd.concat(x[::order], 1)


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:            House Price   R-squared:                       0.735
Model:                            OLS   Adj. R-squared:                  0.704
Method:                 Least Squares   F-statistic:                     23.55
Date:                Sun, 25 Dec 2022   Prob (F-statistic):           1.26e-05
Time:                        16:29:01   Log-Likelihood:                -258.49
No. Observations:                  20   AIC:                             523.0
Df Residuals:                      17   BIC:                             526.0
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
========================================================================================
                           coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------
const                -9.654e+06   5.21e+06     -1.852      0.081   -2.07e+07    1.34e+06
Year of Construction  4960.9407   2607.443      1.903      0.074    -540.283    1.05e+04
House Size (sq.ft.)    394.0417     61.098      6.449      0.000     265.137     522.947
==============================================================================
Omnibus:                        2.064   Durbin-Watson:                   1.926
Prob(Omnibus):                  0.356   Jarque-Bera (JB):                1.689
Skew:                          -0.663   Prob(JB):                        0.430
Kurtosis:                       2.480   Cond. No.                     5.36e+05
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 5.36e+05. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

#Key Notes:



> From the above set of regression, P-value for "Year of Construction" is always high (>5%). So, we can infer "Year of Construction"might not be a good predictor of House Price.


> Number of Rooms and House price may be good predictor (P-Value<5%). 
but when we run regression using the combinamtion the p-value shows high result (11.7% and 62.9%) 


> This indicates that, there we need to include other explanatory variables sush as "Distance from nearest Market", "Road and Transport facilities" etc to predict the House Price more accurately. 





